subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1842,8 +1846,12 @@ static inline int ztls_poll_offload(struct pollfd *fds, int nfds, int timeout)
 			fds[i].revents = 0;
 		}
 
+		(void)sys_mutex_lock(lock, K_FOREVER);
```

Comment:
We should not lock the mutex for the offloaded `poll()` call, just as in the native socket implementation.